roofit/roofit/src/RooMomentMorph.cxx

```diff
@@ -59,32 +57,25 @@ RooMomentMorph::RooMomentMorph(const char *name, const char *title,
   _useHorizMorph(true)
 {
   // observables
-  TIterator* varItr = varList.createIterator() ;
-  RooAbsArg* var ;
-  for (Int_t i=0; (var = (RooAbsArg*)varItr->Next()); ++i) {
+  for (auto *var : varList) {
     if (!dynamic_cast<RooAbsReal*>(var)) {
       coutE(InputArguments) << "RooMomentMorph::ctor(" << GetName() << ") ERROR: variable " << var->GetName() << " is not of type RooAbsReal" << endl ;
       throw string("RooPolyMorh::ctor() ERROR variable is not of type RooAbsReal") ;
     }
     _varList.add(*var) ;
   }
-  delete varItr ;
-
+  
   // reference p.d.f.s
-  TIterator* pdfItr = pdfList.createIterator() ;
-  RooAbsPdf* pdf ;
-  for (Int_t i=0; (pdf = dynamic_cast<RooAbsPdf*>(pdfItr->Next())); ++i) {
+
+  for (auto const *pdf : static_range_cast<RooAbsPdf*>(pdfList)) {
```

Comment:
```suggestion
  for (auto const *pdf : dynamic_range_cast<RooAbsPdf*>(pdfList)) {
```
In the next line there is an explicit check if the `dynamic_cast` succeeded, so you can't use `static_range_cast` here because the pointer would be invalid instead of `nullptr` if the cast is not appropriate!

Yes, you can often replace `dynamic_cast` with `static_cast` in RooFit code to reduce overhead, but not if the cast result is explicitly checked to be zero.